src/services/car/messageLead.ts

```diff
@@ -245,3 +246,75 @@ export const resendMessageLead = async ({
     result: {},
   }
 }
+
+export const fetchDealerWhatsAppLeads = async ({
+  dealerId,
+  query,
+  options = {},
+}: {
+  dealerId: number
+  query: LeadQueryParams
+  options?: ApiCallOptions & { validateOnly?: boolean }
+}): Promise<Paginated<SearchWhatsAppLead>> => {
+  const { validateOnly, ...otherOptions } = options
+  const { page, size, sort = {}, searchQuery } = query
+
+  const { sortOrder, sortType } = sort
+
+  const sortOrDefault = {
+    sortType: sortType || defaultLeadSort.sortType,
+    sortOrder: sortOrder || defaultLeadSort.sortOrder,
+  }
+
+  const queryParams = {
+    page: pageOrDefault(page, defaultDealerMessageLeadsPagination),
+    size: sizeOrDefault(size, defaultDealerMessageLeadsPagination),
+    sort: `${toCamelCase(sortOrDefault.sortType)},${toCamelCase(
+      sortOrDefault.sortOrder
+    )}`,
+    q:
+      searchQuery && searchQuery.length >= 3
+        ? encodeURIComponent(searchQuery)
+        : "",
+  }
+
+  const path = `dealers/${dealerId}/whats-app-tracking-entries?${toQueryString(
+    queryParams
+  )}`
+
+  return await fetchPath({
```

Comment:
I think the `await` is not needed here, no?

Reply:
it needed since `fetchPath` is async function

Reply:
I'm pretty sure you can safely do `return fetchPath()`. Because the promise is already awaited inside fetchPath. Also, all other functions except the messageLeads are returning it directly. Is it somehow different for your use case?